delta_go/game_mechanics.py

```diff
@@ -0,0 +1,265 @@
+        N = self.state.board.shape[0]
+        if move == N**2:
+            return "passes"
+        return f"places counter at coordinate: {(move//N, move%N)}"
+
+    def __str__(self) -> str:
+        return str(self.state.board) + "\n"
+
+    def _step(self, move: int) -> None:
+
+        if self.verbose:
+            name = "player" if self.state.to_play == self.player_color else "opponent"
+            print(f"{name} {self.move_to_string(move)}")
+
+        assert not self.done, "Game is done! Please reset() the env before calling step() again"
+        assert is_move_legal(
+            int_to_coord(move), self.state.board, self.state.ko
+        ), f"{move} is an illegal move"
+
+        self.state = transition_function(self.state, move)
+
+        if self.render:
+            self._render_game()
+
+    def step(self, move: int) -> Tuple[State, int, bool, Dict]:
+
+        assert self.state.to_play == self.player_color
+        self._step(move)
+
+        if not self.done:
+            self._step(self.opponent_choose_move(state=self.state))
+
+        if self.verbose and self.done:
+            self.nice_prints()  # Probably not needed
+
+        return self.state, self.reward, self.done, {}
+
+    def nice_prints(self):
+        print(
+            f"\nGame over. Reward = {self.reward}.\n"
+            f"Player was playing as {self.color_str}\n"
+            f"Black has {np.sum(self.state.board==1)} counters.\n"
+            f"White has {np.sum(self.state.board==-1)} counters.\n"
+            f"Your score is {score(self.state.board, KOMI)}.\n"
+        )
+
+
+def choose_move_randomly(state: State) -> int:
+    return random.choice(all_legal_moves(state.board, state.ko))
+
+
+def choose_move_pass(state: State) -> int:
+    """Always pass."""
+    return PASS_MOVE
+
+
+def load_pkl(team_name: str, network_folder: Path = HERE) -> nn.Module:
+    net_path = network_folder / f"{team_name}_file.pkl"
+    assert (
+        net_path.exists()
+    ), f"Network saved using TEAM_NAME='{team_name}' doesn't exist! ({net_path})"
+    with open(net_path, "rb") as handle:
+        file = pickle.load(handle)
+    return file
+
+
+def save_pkl(file: Any, team_name: str) -> None:
+    assert "/" not in team_name, "Invalid TEAM_NAME. '/' are illegal in TEAM_NAME"
+    net_path = HERE / f"{team_name}_file.pkl"
+    n_retries = 5
+    for attempt in range(n_retries):
+        try:
+            with open(net_path, "wb") as handle:
+                pickle.dump(file, handle, protocol=pickle.HIGHEST_PROTOCOL)
+            load_pkl(team_name)
+            return
+        except Exception:
+            if attempt == n_retries - 1:
+                raise
+
+
+# Need to know the default screen size from petting zoo to get which square is clicked
+# Will not work with a screen override
+SQUARE_SIZE = SCREEN_SIZE[0] // BOARD_SIZE
+LEFT = 1
+
+
+def pos_to_coord(pos: Tuple[int, int]) -> Tuple[int, int]:  # Assume square board
```

Comment:
Docstring would be useful here - I believe users don't need to know what this does, right? So perhaps saying that you don't need to use this might be beneficial?